flink-table-store-core/src/main/java/org/apache/flink/table/store/file/catalog/CatalogFactory.java

```diff
@@ -34,9 +34,9 @@ public interface CatalogFactory {
 
     String identifier();
 
-    Catalog create(String warehouse, ReadableConfig options);
+    Catalog create(String warehouse, Configuration options);
 
-    static Catalog createCatalog(ReadableConfig options) {
+    static Catalog createCatalog(Configuration options) {
         // manual validation
```

Comment:
Now that we have a dedicated `CatalogOptions` class to hold all options, we can call the `validate` method and add those options into required or optional methods.

Reply:
I think we can let it go
If there is a situation where you need to configure some HiveClient parameters, you can do so through this channel.
At the beginning we can be a little more flexible